src/assert.ts

```diff
@@ -29,7 +29,8 @@ export class AssertionError extends Error {
 
 /** https://docs.tsafe.dev/assert */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function assert<_T extends true>(condition: any = true, msg?: string): asserts condition {
```

Comment:
The problem with this line, is that it compiles down into:
`if (condition === undefined) { condition = true }` 

which then doesn't match the type level assertion